src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,15 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
     private final int[] decodeInts(final String string) {
-        final int nValues = ParsingUtils.split(string, INT_DECODE_ARRAY, ',');
+        // find number of CSV items to set array size
+        int numItems = StringUtils.countMatches(string, ",") + 1;
+        String[] intDecodeArray = new String[numItems];
+        final int nValues = ParsingUtils.split(string, intDecodeArray, ',');
```

Comment:
This is from HTSJDK's own class, ParsingUtils, not Apache commons. It also does not use regexes.